src/best-practices/caddy.rst

```diff
@@ -0,0 +1,167 @@
+``http://localhost:5984/db1/doc1``.
+
+.. code-block:: text
+
+    domain.com {
+
+        import /path/to/other/config.caddy # logging, error handling etc.
+
+        proxy /couchdb localhost:5984 {
+            transparent
+            without /couchdb
+        }
+
+    }
+
+Note that in the above configuration, the *Verify Installation* link in
+Fauxton may not succeed.
+
+Reverse proxying + load balancing for CouchDB clusters
+======================================================
+
+Here's a basic excerpt from a Caddyfile in
+``/<path>/<to>/<site>/Caddyfile``. This will proxy and evenly distribute all
+requests from ``http(s)://domain.com/...`` among 3 CouchDB cluster nodes
+at ``localhost:15984``, ``localhost:25984`` and ``localhost:35984``.
+
+Caddy will check the status, i.e. health, of each node every 5 seconds;
+if a node goes down, Caddy will avoid proxying requests to that node until it
+comes back online.
+
+.. code-block:: text
+
+    domain.com {
+
+        import /path/to/other/config.caddy # logging, error handling etc.
+
+        proxy / http://localhost:15984 http://localhost:25984 http://localhost:35984 {
+            policy round_robin
+            health_check /_up
+            health_check_duration 5s
+            try_interval 500ms
+            timeout 1h
+            transparent
+        }
+
+    }
+
+Note that in the above configuration, the *Verify Installation* link in
+Fauxton may not succeed.
+
+Authentication with Caddy as a reverse proxy
+============================================
+
+Here's a sample config setting with basic authentication enabled, placing
+CouchDB in the ``/couchdb`` subdirectory:
+
+.. code-block:: text
+
+    domain.com {
+
+        import /path/to/other/config.caddy # logging, error handling etc.
+
+        basicauth /couchdb couch_username couchdb_password
+
+        proxy /couchdb localhost:5984 {
+            transparent
+            header_upstream -Authorization
+            without /couchdb
+        }
+
+    }
+
+For security reasons, using a plaintext password in the ``Caddyfile`` is not
+advisable. One solution is to define Caddy-process environment variables e.g.
+``COUCH_PW=couchdb_password`` and using placeholders in the ``Caddyfile``
+instead, e.g. ``{$COUCH_PW}``.
+
+This setup leans entirely on Caddy performing authorization, and forwarding
+requests to CouchDB with no authentication (with CouchDB in Admin Party mode).
+For a better solution, see :ref:`api/auth/proxy`.
+
+SSL/TLS with Caddy
+==================
+
+Caddy is http-by-default, and will automatically acquire, install, activate and,
+when necessary, renew a trusted SSL certificate for you - all in the background.
+Certificates are issued by the LetsEncrypt certificate authority.
+
+.. code-block:: text
+
+    domain.com {
+
+        import /path/to/other/config.caddy # logging, error handling etc.
+
+        proxy / localhost:5984 {
+            transparent
+            header_upstream x-forwarded-ssl on
```

Comment:
What about the `X-Forwarded-For` and `Host` headers, are these automatic? Same comment as earlier in this file. Replications will likely fail without these set correctly.

Reply:
Clarified with a note under Basic Config: https://github.com/apache/couchdb-documentation/pull/318/commits/dfaa2c6b286e036f8a8677447138a4c0dfb1ee6c